src/main/java/com/google/codeu/servlets/LogoDetectServlet.java

```diff
@@ -0,0 +1,109 @@
+package com.google.codeu.servlets;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/logo-detect")
+public class LogoDetectServlet extends HttpServlet {
+
+  JsonArray logoJsonArray;
+
+  @Override
+  public void init() {
+    this.logoJsonArray = new JsonArray();
+    //this.logoJsonArray.add("starbucks");
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    response.getOutputStream().println(this.logoJsonArray.toString());
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
+    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
+    List<BlobKey> blobKeys = blobs.get("image");
+
+    if (blobKeys != null && !blobKeys.isEmpty()) {
+      BlobKey blobKey = blobKeys.get(0);
+      ImagesService imagesService = ImagesServiceFactory.getImagesService();
+      ServingUrlOptions options = ServingUrlOptions.Builder.withBlobKey(blobKey);
+      String imageUrl = imagesService.getServingUrl(options);
+
+      byte[] blobBytes = getBlobBytes(blobstoreService, blobKey);
+      String logos = getLogoInformation(blobBytes);
+      String[] logosArray = logos.split(",");
+      this.logoJsonArray.add(logosArray[0]); //Chose the first logo for now
```

Comment:
I think I see now where the main confusion is. Your logoJsonArray is a global object that exists on a servlet level. The servlet can potentially serve many requests at the same time, so there is no way for you to guarantee that the user calling doGet (indirectly) of this servlet is the same user who earlier called "doPost". You need to find a way to pass this information to `/info-present.html`, which does not involve using the servlet object data.